### PR TITLE
Add a switch to disable tail call optimisations on x86_64.

### PR DESCRIPTION
**Pull request by vext01**
This is a companion PR to https://github.com/ykjit/yk/pull/504

Works around https://github.com/ykjit/yk/issues/502

### CODE REVIEW

File: llvm/lib/Target/X86/X86ISelLowering.cpp [hunk 1]
```diff
@@ -3918,6 +3923,16 @@ X86TargetLowering::LowerCall(TargetLowering::CallLoweringInfo &CLI,
   bool isVarArg                         = CLI.IsVarArg;
   const auto *CB                        = CLI.CB;
 
+  // Tail call optimisations interfere with the trace compiler's inlining
+  // stack, as we never see the return for the caller (the caller's frame is
+  // reused, and the callee's return returns from both the caller and the
+  // callee).
+  //
+  // FIXME: https://github.com/ykjit/yk/issues/502
```
Comment (ltratt) [inline llvm/lib/Target/X86/X86ISelLowering.cpp:3931]: Should this be a `YKFIXME`?

Comment (vext01) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:3931]: Yes it should!

Comment (vext01) [reply llvm/lib/Target/X86/X86ISelLowering.cpp:3931]: Fixed in e8239fa1956a

**Pull request merged**